### PR TITLE
Fix boolean for empty objects

### PR DESCRIPTION
**Pull request by dmitrika**
Fix for #1342

It's my first pr for Prettier 😅 So, I'm not sure if I've done all right.

Seems, like I've break `tests/arrays/__snapshots__/jsfmt.spec.js.snap` with oneline expression. Should I add check like `node.left.type !== "Identifier"` ?

### CODE REVIEW

File: tests/arrays/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -4,8 +4,10 @@ exports[`empty.js 1`] = `
 const a = someVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeLong.Expression || [];
 const a = someVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeLong.Expression || {};
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-const a = someVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeLong.Expression || [];
-const a = someVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeLong.Expression || {};
+const a =
+  someVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeLong.Expression || [];
+const a =
+  someVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeLong.Expression || {};
```
Comment (josephfrazier) [inline tests/arrays/__snapshots__/jsfmt.spec.js.snap]: Huh, it looks like this wasn't honoring the `printWidth` before. Nicely done!

Comment (josephfrazier) [inline tests/arrays/__snapshots__/jsfmt.spec.js.snap]: Huh, it looks like this wasn't honoring the maximum line length before. Nicely done!

File: src/printer.js [hunk 2]
```diff
@@ -3678,11 +3678,17 @@ function isBinaryish(node) {
 }
 
 function shouldInlineLogicalExpression(node) {
-  return (
-    node.type === "LogicalExpression" &&
-    (node.right.type === "ObjectExpression" ||
-      node.right.type === "ArrayExpression")
-  );
+  if (node.type === "LogicalExpression" &&
```
Comment (vjeux) [inline src/printer.js]: Can you early return false if that's a logical expression, this way you don't need to have it written twice.

File: src/printer.js [hunk 3]
```diff
@@ -3678,11 +3678,17 @@ function isBinaryish(node) {
...
+      node.right.type === "ObjectExpression") {
+    return node.right.properties.length !== 0;
```
Comment (vjeux) [inline src/printer.js]: Would be nice to put this complex expression inside of the if condition and return true. It's easier to understand.

File: tests/binary-expressions/__snapshots__/jsfmt.spec.js.snap [hunk 4]
```diff
@@ -242,6 +244,14 @@ const x =
   (longVariable && longVariable) ||
   (longVariable && longVariable) ||
   (longVariable && longVariable);
+const x =
+  firstItemWithAVeryLongNameThatKeepsGoing ||
+  firstItemWithAVeryLongNameThatKeepsGoing ||
+  {};
```
Comment (vjeux) [inline tests/binary-expressions/__snapshots__/jsfmt.spec.js.snap]: Can you make sure that the following pattern still have `[]` on the same line?

```js
const x =
  call(
    reallyBigOne,
    reallyBigOne,
  ) || [];
```

File: tests/arrays/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (vjeux) [inline tests/arrays/__snapshots__/jsfmt.spec.js.snap]: This was actually intentional, I wanted to make sure that the following pattern ( https://github.com/prettier/prettier/pull/1590#discussion_r116126351 ) is printed in one line. The pattern in this comment is almost never used so it doesn't really matter if it is > 80 columns. Also putting it on the next line isn't really pretty either.

File: tests/binary-expressions/__snapshots__/jsfmt.spec.js.snap (hunk 4 above)
Comment (dmitrika) [inline tests/binary-expressions/__snapshots__/jsfmt.spec.js.snap]: Yes, added this case, it's printed like in your example.

**Comment by vjeux** [comment]
Thanks!

**Pull request merged**